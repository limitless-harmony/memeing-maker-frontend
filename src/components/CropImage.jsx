import React from 'react';
import styled from 'styled-components';
import Cropper from 'react-easy-crop';

const CropImage = ({
  image,
  crop,
  zoom,
  aspect,
  onCropChange,
  onCropComplete,
  onZoomChange,
}) => (
  <StyledCropImage>
    <Cropper
      image={image}
      crop={crop}
      zoom={zoom}
      aspect={aspect}
      onCropChange={onCropChange}
      onCropComplete={onCropComplete}
      onZoomChange={onZoomChange}
    />
  </StyledCropImage>
);

const StyledCropImage = styled.div`
  position: relative;
  width: 100%;
  height: 400px;
`;

export default CropImage;
