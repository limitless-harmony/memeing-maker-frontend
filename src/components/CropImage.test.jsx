import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import CropImage from 'components/CropImage';

describe('Testing the CropImage component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<CropImage />);
    expect(wrapper).toMatchSnapshot();
  });
});
