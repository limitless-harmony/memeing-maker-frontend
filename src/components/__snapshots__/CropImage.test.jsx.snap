// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CropImage component matches snapshot 1`] = `
<styled.div>
  <Cropper
    aspect={1.3333333333333333}
    classes={Object {}}
    cropShape="rect"
    maxZoom={3}
    minZoom={1}
    showGrid={true}
    style={Object {}}
    zoom={1}
    zoomSpeed={1}
  />
</styled.div>
`;
